packages/mui-codemod/src/deprecations/button-classes/postcss-plugin.js

```diff
@@ -0,0 +1,190 @@
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedPrimary',
+    replacementSelector: '.MuiButton-contained.MuiButton-colorPrimary',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedSecondary',
+    replacementSelector: '.MuiButton-contained.MuiButton-colorSecondary',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedSuccess',
+    replacementSelector: '.MuiButton-contained.MuiButton-colorSuccess',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedError',
+    replacementSelector: '.MuiButton-contained.MuiButton-colorError',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedInfo',
+    replacementSelector: '.MuiButton-contained.MuiButton-colorInfo',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedWarning',
+    replacementSelector: '.MuiButton-contained.MuiButton-colorWarning',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-textSizeSmall',
+    replacementSelector: '.MuiButton-text.MuiButton-sizeSmall',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-textSizeLarge',
+    replacementSelector: '.MuiButton-text.MuiButton-sizeLarge',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-outlinedSizeSmall',
+    replacementSelector: '.MuiButton-outlined.MuiButton-sizeSmall',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-outlinedSizeLarge',
+    replacementSelector: '.MuiButton-outlined.MuiButton-sizeLarge',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedSizeSmall',
+    replacementSelector: '.MuiButton-contained.MuiButton-sizeSmall',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedSizeLarge',
+    replacementSelector: '.MuiButton-contained.MuiButton-sizeLarge',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-textSizeMedium',
+    replacementSelector: '.MuiButton-text.MuiButton-sizeMedium',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-outlinedSizeMedium',
+    replacementSelector: '.MuiButton-outlined.MuiButton-sizeMedium',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-containedSizeMedium',
+    replacementSelector: '.MuiButton-contained.MuiButton-sizeMedium',
+    replacementSelectorPrefix: '&.',
+  },
+  {
+    deprecatedClass: '.MuiButton-iconSizeSmall',
+    replacementSelector: '.MuiButton-icon.MuiButton-sizeSmall',
+    replacementSelectorPrefix: '& .',
+  },
+  {
+    deprecatedClass: '.MuiButton-iconSizeMedium',
+    replacementSelector: '.MuiButton-icon.MuiButton-sizeMedium',
+    replacementSelectorPrefix: '& .',
```

Comment:
Shouldn't the replacement be `&.MuiButton-sizeMedium .MuiButton-icon`?

Reply:
from my understanding, `iconSizeMedium` is split into `icon` and `sizeMedium`, both these classes are applied to `icon` slot and not `root` slot. But your suggestion seems to add `sizeMedium` class to root slot. what do you think

Reply:
The `size` classes are applied to the root: https://mui.com/material-ui/api/button/#button-classes-sizeLarge

So to replace the `.MuiButton-iconSizeMedium` class, we need to select `.MuiButton-sizeMedium .MuiButton-icon`: An element with class `MuiButton-icon` within an element with class `MuiButton-sizeMedium`. We might want to use the direct child selector `.MuiButton-sizeMedium > .MuiButton-icon`.